node/src/components/consensus/highway_core/state.rs

```diff
@@ -270,6 +270,7 @@ impl<C: Context> State<C> {
 
     pub(crate) fn add_evidence(&mut self, evidence: Evidence<C>) {
         let idx = evidence.perpetrator();
+        info!(?evidence, "marking validator #{} as faulty", idx.0);
```

Comment:
I would log also the `SignedWireVote`s that create equivocation.

Reply:
They are part of the `evidence`.

Reply:
Ah, missed that :see_no_evil: